File: invert-binary-tree/InvertBinaryTree.swift
public class TreeNode {
    public var val: Int
    public var left: TreeNode?
    public var right: TreeNode?
    public init(_ val: Int) {
        self.val = val
        self.left = nil
        self.right = nil
    }
}

// recursive
func invertTree(_ root: TreeNode?) -> TreeNode? {
    let temp = root?.left
    root?.left = root?.right
    if let left = root?.left {
        root?.left = invertTree(left)
    }
    root?.right = temp
    if let right = root?.right {
        root?.right = invertTree(right)
    }
    return root
}

File: invert-binary-tree/InvertBinaryTree.java
import java.util.LinkedList;
import java.util.Queue;

public class InvertBinaryTree {
  public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
  }

  public TreeNode invertTree(TreeNode root) {
    if (root == null) {
      return null;
    }

    Queue<TreeNode> queue = new LinkedList<>();
    queue.add(root);
    while (!queue.isEmpty()) {
      TreeNode r = queue.poll();
      TreeNode temp = r.right;
      r.right = r.left;
      r.left = temp;
      if (r.right != null) {
        queue.add(r.right);
      }
      if (r.left != null) {
        queue.add(r.left);
      }
    }
    return root;
  }
}

